apps/agent/pkg/ratelimit/fixed_window.go

```diff
@@ -0,0 +1,109 @@
+package ratelimit
+
+import (
+	"sync"
+	"time"
+)
+
+type identifier struct {
```

Comment:
This is a bit wasteful in terms of allocations, but we have such little load right now